include/NovelRT/Audio/AudioService.h

```diff
@@ -26,38 +26,45 @@ namespace NovelRT::Audio
         MusicBank _music;
         ALuint _musicSource;
         ALint _musicSourceState;
+        bool _musicStopRequested;
         ALint _musicLoopAmount;
         ALint _soundLoopAmount;
         ALint _soundSourceState;
         SoundBank _soundStorage;
         SoundBank _bufferStorage;
 
-        ALuint readFile(std::string input);
-        std::string getALError();
+        ALuint ReadFile(std::string input);
+        std::string GetALError();
 
     public:
         bool isInitialised;
 
         AudioService();
         ~AudioService();
 
-        bool initializeAudio();
-        std::vector<ALuint>::iterator loadMusic(std::string input);
-
-        void setSoundVolume(ALuint source, float val);
-        void setSoundPosition(ALuint source, float posX, float posY);
-        void resumeMusic();
-        void playMusic(std::vector<ALuint>::iterator handle, int32_t loops);
-        void pauseMusic();
-        void stopMusic();
-        void setMusicVolume(float value);
-        void checkSources();
-        ALuint loadSound(std::string input);
-        void unload(ALuint handle);
-        void playSound(ALuint handle, int32_t loops);
-        void stopSound(ALuint handle);
-        void tearDown();
+        bool InitializeAudio();
+        std::vector<ALuint>::iterator LoadMusic(std::string input);
+
+        void SetSoundVolume(ALuint source, float val);
+        void SetSoundPosition(ALuint source, float posX, float posY);
+        void ResumeMusic();
+        void PlayMusic(std::vector<ALuint>::iterator handle, int32_t loops);
+        void PauseMusic();
+        void StopMusic();
+        void SetMusicVolume(float value);
+        void CheckSources();
+        ALuint LoadSound(std::string input);
+        void Unload(ALuint handle);
+        void PlaySound(ALuint handle, int32_t loops);
+        void StopSound(ALuint handle);
+        void TearDown();
+        [[nodiscard]] bool IsLoaded(std::vector<ALuint>::iterator handle);
+        [[nodiscard]] bool IsLoaded(ALuint handle);
+        [[nodiscard]] bool IsMusicPlaying();
+        [[nodiscard]] bool IsSoundPlaying(ALuint handle);
+        float GetMusicVolume();
+        float GetSoundVolume(ALuint handle);
```

Comment:
these should be [[nodiscard]] ideally. ☺️

Reply:
true, true